Tests/NetworkServiceTests/NetworkServiceTests.swift

```diff
@@ -1,8 +1,170 @@
-import NetworkService
 import XCTest
+import NetworkService
 
 final class NetworkServiceTests: XCTestCase {
-    func testUseMethodExistsInAPI() {
-        NetworkService().testUse()
+    
+    var networkService: NetworkService<MockEndpoint>!
+    
+    override func setUpWithError() throws {
+        networkService = NetworkService()
+        URLProtocol.registerClass(MockURLProtocol.self)
+    }
+    
+    override func tearDownWithError() throws {
+        URLProtocol.unregisterClass(MockURLProtocol.self)
+        networkService = nil
+    }
+    
+    func testSuccessfulRequestReturnsExpectedData() async throws {
+        // Given
+        let expectedData = try XCTUnwrap("Mocked Response Data".data(using: .utf8))
+        MockURLProtocol.requestHandler = { request in
+            let response = try XCTUnwrap(HTTPURLResponse(url: XCTUnwrap(request.url), statusCode: 200, httpVersion: nil, headerFields: nil))
+            return (response, expectedData)
+        }
+        
+        let endpoint = MockEndpoint()
+        
+        // When
+        let data = try await networkService.request(endpoint)
+        
+        // Then
+        XCTAssertEqual(data, expectedData)
+    }
+    
+    func testBadRequestStatusCodeInResponseTriggersBadRequestError() async throws {
+        // Given
```

Comment:
I think it's more appropriate.
```suggestion
        // Given // When
```